### ISSUE TITLE
Service instability

### ISSUE THREAD

**Issue by henriesteves** [SchneeHertz/node-edge-tts#4]
I've been using this package for a while now. It's been a few days since it seems to be unstable. Sometimes it generates the audio file, sometimes it doesn't. Can you tell me if there have been any changes to the Microsoft API? Did you notice any problems too?

**Comment by danilocanalle** [comment]
Same here.

Apparently Microsoft included new voices, they must have changed something in the authentication.

**Comment by SchneeHertz** [comment]
I tested it several times and didn't find any instances of generation failure. Is there any other error message?

**Comment by henriesteves** [comment]
I tested a little more here and realized that the instability is in the Brazilian Portuguese language that we are using (pt-BR-ThalitaNeural). Testing American English doesn't seem to be causing any problems. In the documentation of the Microsoft API, there were several changes in the Brazilian Portuguese voice, perhaps it is just temporary until they solve some problem. I'll continue testing here, I'll report anything here if I find any errors.

**Comment by danilocanalle** [comment]
True! What I noticed in the tests when it failed is that a promise is started and it doesn't resolve, it remains pending. 

We could have a default timeout to return an error. What do you think?

To simulate you can use the voice: pt-BR-ThalitaNeural

I've created this PR: https://github.com/SchneeHertz/node-edge-tts/pull/5

**Comment by SchneeHertz** [comment]
I merged the PR and updated to version 1.2.4.
You can now set a timeout, but this timeout is effective throughout the entire process of TTS operation. If you need to generate long texts, please extend the timeout value appropriately
This does not solve the issue of service instability, and this issue will remain open for now

**Comment by SchneeHertz** [comment]
Users in mainland China reported a 403 error when connecting, which was caused by the Edge browser changing the connection URL. I updated to 1.2.7 to fix this issue.
Although it behaves differently with this issue, please test after the upgrade to see if the connection situation has improved

**Comment by jianchang512** [comment]
Does this code support deployment to cloudflare workers?

**Comment by SchneeHertz** [comment]
> Does this code support deployment to cloudflare workers?

Please do not ask questions under unrelated issues.

And, No
